csrc/ir/nodes.cpp

```diff
@@ -2361,9 +2361,46 @@ LoadStoreOp::LoadStoreOp(
   addDataAttribute(cache_op);
 }
 
+namespace {
+// Returns the permutation from `in` to `out`, i.e., `out[i]==in[perm[i]]`. As a
+// precondition, `out` must be a permutation of `in` per the definition of
+// std::is_permutation.
+template <typename T>
+std::vector<int64_t> computePermutation(
+    const std::vector<T>& in,
+    const std::vector<T>& out) {
+  std::vector<int64_t> permutation;
+  permutation.reserve(out.size());
+  // O(n^2) is totally fine for the current use case of computing the
+  // root-to-rfactor permutation. If needed, this can be improved by requiring T
+  // to be hashable and/or comparable.
+  for (const T& out_element : out) {
+    permutation.push_back(std::distance(
+        in.begin(), std::find(in.begin(), in.end(), out_element)));
+  }
+  return permutation;
+}
+} // namespace
+
 std::vector<PolymorphicValue> LoadStoreOp::evaluate(
     const ExpressionEvaluator& ee,
     const std::vector<PolymorphicValue>& inputs) const {
+  if (TensorView* out_tv = dynamic_cast<TensorView*>(out())) {
```

Comment:
How many `op_types` are there for `LoadStoreOp`. Should we split the evaluate method (for instance: a switch-case implementation) based on `op_type`?

Reply:
I remember several. According to https://github.com/NVIDIA/Fuser/blob/a55c2df1d45f0fb5030b068ab7755d0527163e6b/csrc/ir/utils.cpp#L576, only those with an rfactor domain are used for permutation and others can simply be implemented as input forwarding. So I think it's better to not split based on `op_type`. Wdyt?

Reply:
Got it, this is fine then.